### ISSUE TITLE
How can we help CSP gain more adoption with Web Developers

### ISSUE THREAD

**Issue by torgo** [w3c-cg/swag#3]
There was a paper from 2020 https://publications.cispa.saarland/2986/1/roth2020csp.pdf (ref from @simoneonofri). There's [documentation out there](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) e.g. on MDN. There are [tools out there](https://csp-evaluator.withgoogle.com/). So what is missing to help CSP gain more adoption with web developers?

**Comment by torgo** [comment]
Also noting that there is CSP content in the 121 free course that OpenSSF provides: https://training.linuxfoundation.org/training/developing-secure-software-lfd121/ Maybe that could be one way to help drive CSP adoption.

**Comment by simoneonofri** [comment]
Yes, documentation and training are both important. Maybe we can use W3Cx, too. @marieforgue, can you explain how it works?

**Comment by aaronshim** [comment]
Hi! Perhaps a philosophical question around how this issue title is phrased-- have we all decided that having courses is the best way forward for getting more developer mind share for adopting CSP?

I think a course is a wonderful idea, but at the same time, I wonder what other ideas we can throw on the wall here-- for instance, one that I would like to see happen is increasing the number of **frameworks** that make it easy to have a low-to-no-config safe-by-default CSP enforcement option to simplify some of the complexity (that we need a course to clarify).

**Comment by simoneonofri** [comment]
> Hi! Perhaps a philosophical question around how this issue title is phrased-- have we all decided that having courses is the best way forward for getting more developer mind share for adopting CSP?
> 
> I think a course is a wonderful idea, but at the same time, I wonder what other ideas we can throw on the wall here-- for instance, one that I would like to see happen is increasing the number of **frameworks** that make it easy to have a low-to-no-config safe-by-default CSP enforcement option to simplify some of the complexity (that we need a course to clarify).

I think you is a wonderful idea, framework and education as a "pincer".

I am collecting some feedback from the broader community: there are often inline things and developers needs to understand how to manage in a seamless way (no-code?) [the indicator is when we found unsafe-inline and unsafe-eval] and how to hash quickly the scripts.
